src/client_conn.c

```diff
@@ -54,55 +56,50 @@ int get_socket() {
 
 static void tetris_translate(void *net_client, int x) {
 	char xdir = x > 0 ? 1 : 0;
-	char message[128];
-	sprintf(message, "%c%c", MSG_TYPE_TRANSLATE, xdir);
-	tetris_send_message((NetClient *)net_client, message);
+	char message[2];
+	sprintf(message, "%c", xdir);
+	tetris_send_message((NetClient *)net_client, message,
+	                    MSG_TYPE_TRANSLATE);
 }
 
 static void tetris_lower(void *net_client) {
-	char message[8];
-	sprintf(message, "%c", MSG_TYPE_LOWER);
-	tetris_send_message((NetClient *)net_client, message);
+	ttetris_net_request((NetClient *)net_client, NULL, 0, MSG_TYPE_LOWER);
 }
 
 static void tetris_rotate(void *net_client, int theta) {
 	char dir = theta > 0 ? 1 : 0;
-	char message[128];
-	sprintf(message, "%c%c", MSG_TYPE_ROTATE, dir);
-	tetris_send_message((NetClient *)net_client, message);
+	char message[2];
+	sprintf(message, "%c", dir);
+	tetris_send_message((NetClient *)net_client, message, MSG_TYPE_ROTATE);
 }
 
 static void tetris_drop(void *net_client) {
-	char message[128];
-	sprintf(message, "%c", MSG_TYPE_DROP);
-	tetris_send_message((NetClient *)net_client, message);
+	ttetris_net_request((NetClient *)net_client, NULL, 0, MSG_TYPE_DROP);
 }
 
 static void tetris_swap_hold(void *net_client) {
-	char message[128];
-	sprintf(message, "%c", MSG_TYPE_SWAP_HOLD);
-	tetris_send_message((NetClient *)net_client, message);
+	ttetris_net_request((NetClient *)net_client, NULL, 0,
+	                    MSG_TYPE_SWAP_HOLD);
 }
 
 StringArray *tetris_list(NetClient *net_client) {
-	char message[1];
-	message[0] = MSG_TYPE_LIST;
-	NetRequest *request = ttetris_net_request(net_client, message, 1);
+	NetRequest *request =
+	    ttetris_net_request(net_client, "", 0, MSG_TYPE_LIST);
 	ttetris_net_request_block_for_response(request);
 
 	MessageHeader *header = (MessageHeader *)request->cursor;
 
 	// deserialize names
 	Blob *body = malloc(sizeof(Blob));
-	body->bytes = request->cursor + 1;
+	body->bytes = request->cursor;
```

Comment:
Curious, what changed to not require this `+ 1`?

Reply:
Yeah; this is kind of confusing. It gets into the nitty-gritty of our message wire format.

The reason is because the first byte of the **message body** used to indicate the type of message (one of the `#declare`s in header.h such as `MSG_TYPE_REGISTER`). In this patch, I **moved the message type into MessageHeader**, which defines the first few bytes of every message.

Since the message type has been removed from the body, we no longer need to increment the cursor past it (The cursor is automatically pushed to the start of the body by logic in `read_from_client` and `read_from_server`).

In the future, I'd love to document our wire format a little so it's easy to understand when people find our repo. It's evolved a lot since the first version.

Reply:
If you're curious, here is the [corresponding server change](https://github.com/Nitepone/terminally-tetris/pull/30/files/88a277429027f9bbede8ecf12556cdd087616809#diff-cbf91a87d3f07d2ad455ab70d64c16f55b7dc82be33346014bf25f91296d1e9eL40-L50) that removes the extra byte from the message body.

Reply:
I see! Makes sense to me!